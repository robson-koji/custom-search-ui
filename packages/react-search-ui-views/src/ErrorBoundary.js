import PropTypes from "prop-types";
import React from "react";

import { appendClassName } from "./view-helpers";

function ErrorBoundary({ children, className, error }) {
  if (error) {
    return (
      <div className={appendClassName("sui-search-error", className)}>
        {error}
      </div>
    );
  }

  return children;
}

ErrorBoundary.propTypes = {
  children: PropTypes.node.isRequired,
  error: PropTypes.string.isRequired,
  className: PropTypes.string
};

export default ErrorBoundary;
