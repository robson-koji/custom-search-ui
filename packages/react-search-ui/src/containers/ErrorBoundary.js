import PropTypes from "prop-types";
import React, { Component } from "react";
import { withSearch } from "..";
import { ErrorBoundary } from "@elastic/react-search-components";

export class ErrorBoundaryContainer extends Component {
  static propTypes = {
    children: PropTypes.node.isRequired,
    error: PropTypes.string.isRequired
  };

  render() {
    return <ErrorBoundary {...this.props} />;
  }
}

export default withSearch(ErrorBoundaryContainer);
