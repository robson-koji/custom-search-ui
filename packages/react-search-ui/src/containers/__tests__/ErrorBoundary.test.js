import React from "react";
import { shallow } from "enzyme";
import { ErrorBoundaryContainer } from "../ErrorBoundary";

const params = {
  children: <div>Child</div>,
  error: "I am an error"
};

it("renders correctly", () => {
  const wrapper = shallow(<ErrorBoundaryContainer {...params} />);
  expect(wrapper).toMatchSnapshot();
});

it("supports a render prop", () => {
  const render = ({ error }) => {
    return <div>{error}</div>;
  };
  const wrapper = shallow(
    <ErrorBoundaryContainer {...params} render={render} />
  );
  expect(wrapper.find(render).dive()).toMatchSnapshot();
});
