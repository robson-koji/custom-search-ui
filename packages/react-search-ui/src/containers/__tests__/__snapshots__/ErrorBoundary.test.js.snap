// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ErrorBoundary
  error="I am an error"
>
  <div>
    Child
  </div>
</ErrorBoundary>
`;
